python/ray/data/dataset.py

```diff
@@ -1961,41 +1998,99 @@ def to_torch(self,
         from ray.data.impl.torch_iterable_dataset import \
             TorchIterableDataset
 
-        if feature_columns and feature_column_dtypes:
-            if len(feature_columns) != len(feature_column_dtypes):
-                raise ValueError("The lengths of `feature_columns` "
-                                 f"({len(feature_columns)}) and "
-                                 f"`feature_column_dtypes` ("
-                                 f"{len(feature_column_dtypes)}) do not "
-                                 "match!")
+        multi_input = feature_columns and (isinstance(feature_columns, dict) or
+                                           isinstance(feature_columns[0],
+                                                      (list, tuple)))
+
+        if not feature_columns:
+            feature_columns = None
```

Comment:
nit: Would prefer to add a comment here describing this implicit behavior, or just raising an error here requiring the user to explicitly choose between `None` and  non-empty columns.

Reply:
```suggestion
       # If an empty collection is passed in, treat it the same as None
        if not feature_columns:
            feature_columns = None
```